docs/rfcs/0001-video-retranscoding.md

```diff
@@ -0,0 +1,43 @@
+## Video Re-transcoding
+
+#### Abstract
+Provide a way to trigger re-transcoding of specified videos, or all videos in a collection, if new presets are needed.
+
+
+#### Architecture Changes
+
+- Create a flag on Collection and Video: boolean field ‘schedule_retranscode’
+- Create a recurring task to look for videos and collections with `schedule_retranscode=True`.
+  - check flagged collections first:
+     - set `schedule_retranscode=True` for every Video in the Collection, then
+     - set `schedule_retranscode=False` on the Collection
+  - Trigger a new encode job for each flagged Video.
+  - Assign a new video status - ‘Retranscoding’
+  - Set `schedule_retranscode=False` on the Video
+- Modify the generation of encoding job metadata:
+  - a temporary prefix should be added to the output filenames to avoid transcode errors that occur when trying to overwrite existing output from the previous transcode
+- Videos being re-transcoded should still be playable (front end needs to accept the new reencoding status as playable)
+- Modify the task that checks on video transcode statuses
+  - should handle the new Retranscoding status
+  - There will now be 2+ encode jobs for the video instead of just 1, the most recent is the one that needs to be checked for progress.
+- When a retranscoding is complete:
+  - copy output from the temporary S3 subfolder to the correct subfolder, then delete the temporary files
+  - Change video status back to complete
```

Comment:
@gsidebo I think we also need to update edX when the retranscode is complete, since I believe the HLS cloudfront URL will change. Does it make sense to add that to this process as well?